contracts/reserve/ExchangeHelper.sol

```diff
@@ -126,29 +127,29 @@ contract ExchangeHelper is DAOUpgradeableContract {
 		}
 
 		uint256 result;
-		if (address(_buyWith) == cDaiAddress) {
+		if (_buyPath.length == 1 && _buyPath[0] == cDaiAddress) {
```

Comment:
no need for this .length check
add require in the begining that buyPath[last] = dai
same for sell sellPath[0] = dai

Reply:
buyPath[last] = dai this can be also cDAI so we can't check it initially